packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.m

```diff
@@ -211,20 +211,50 @@ - (instancetype)initWithMediaSettings:(FCPPlatformMediaSettings *)mediaSettings
   [_motionManager startAccelerometerUpdates];
 
   if (_mediaSettings.framesPerSecond) {
+    [_mediaSettingsAVWrapper beginConfigurationForSession:_videoCaptureSession];
+
+    // Possible values for presets are hard-coded in FLT interface having
```

Comment:
Why is all this code moving out of the locked section; is that safe?

Reply:
I did not see anything saying that the order of `beginConfiguration` and `lockForConfiguration` is important or that setting the session preset must be under that lock. There is example where it is used in that order: https://developer.apple.com/documentation/avfoundation/avcapturedevice/1389221-activeformat?language=objc

I moved `setCaptureSessionPreset` outside of lock because it can itself call `lockForConfiguration` and `unlockForConfiguration` (and it is also called elsewhere even without `beginConfiguration`).

Reply:
Seems `lockForConfiguration` is actually recursive from my observation (I could not get mentioned exceptions until now) as locking twice and unlocking once does not cause exceptions but unlocking twice yes. So I can revert this change (?).

Also it seems not all `commitConfiguration` calls go through `_mediaSettingsAVWrapper`.